src/scipp/coords.py

```diff
@@ -2,247 +2,480 @@
 # Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock, Jan-Lukas Wynen
 
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from copy import copy
+import dataclasses
+from enum import Enum, auto
+from graphlib import TopologicalSorter
```

Comment:
This does not appear to be a standard library? Can we avoid its use?

Reply:
It was added in 3.9. I forgot to add the backport to the dependencies.

> Can we avoid its use?

Possibly by writing a topological sort. But do you think it is a problem to use the backport until we transition to 3.9?

Reply:
If it is just a backport then it is ok, I think.